cmd/juju/caas/add.go

```diff
@@ -876,34 +876,26 @@ func (c *AddCAASCommand) getClusterMetadataFunc(ctx *cmd.Context) provider.GetCl
 }
 
 func (c *AddCAASCommand) verifyName(name string) error {
-	public, _, err := c.cloudMetadataStore.PublicCloudMetadata()
-	if err != nil {
-		return err
+	if name == k8s.K8sCloudMicrok8s {
+		return fmt.Errorf("%q is the name of a built-in cloud", name)
```

Comment:
Can we also tell the user what to do to avoid the error, eg 
"microk8s is the name of a built-in cloud, please use a different name"

We could also inform the user:
If you want to use Juju with microk8s, the recommended way is to install the strictly confined microk8s snap.
Using the strictly confined microk8s snap means that Juju and microk8s will work together out of the box."